staging/src/k8s.io/apiserver/pkg/admission/plugin/webhook/admission.go

```diff
@@ -245,7 +262,61 @@ func (a *GenericAdmissionWebhook) Admit(attr admission.Attributes) error {
 	return errs[0]
 }
 
+func (a *GenericAdmissionWebhook) getNamespaceLabels(attr admission.Attributes) (map[string]string, error) {
+	// if the request itself is about a namespace
+	if attr.GetResource().Resource == "namespaces" && len(attr.GetSubresource()) == 0 {
+		accessor, err := meta.Accessor(attr.GetObject())
+		if err != nil {
+			return nil, err
+		}
+		return accessor.GetLabels(), nil
+	}
+
+	namespaceName := attr.GetNamespace()
```

Comment:
...and if the object doesn't have a namespace?

Reply:
It's handled before this function is called. Such objects are never exempted.